pkgs/servers/nosql/eventstore/default.nix

```
@@ -1,72 +1,79 @@
-{ lib, stdenv
-, fetchFromGitHub
-, fetchurl
-, makeWrapper
+{ lib
+, git
 , dotnetCorePackages
+, glibcLocales
+, buildDotnetModule
+, fetchFromGitHub
+, bintools
+, stdenv
 , mono
-, Nuget
+, patchelf
 }:
 
-let
-
-  dotnet-sdk = dotnetCorePackages.sdk_5_0;
-
-  deps = import ./deps.nix { inherit fetchurl; };
-
-in
-
-stdenv.mkDerivation rec {
-
+buildDotnetModule rec {
   pname = "EventStore";
-  version = "5.0.8";
+  version = "21.10.5";
 
   src = fetchFromGitHub {
     owner = "EventStore";
     repo = "EventStore";
     rev = "oss-v${version}";
-    sha256 = "021m610gzmrp2drywl1q3y6xxpy4qayn580d855ag952z9s6w9nj";
+    sha256 = "sha256-uUDjTGCiQgXmvOUsujIA0JkGQGuw9U4zLKDP1WIFq1o=";
+    leaveDotGit = true;
   };
 
-  buildInputs = [
-    makeWrapper
-    dotnet-sdk
-    mono
-    Nuget
-  ];
-
-  # that dependency seems to not be required for building, but pulls in libcurl which fails to be located.
-  # see: https://github.com/EventStore/EventStore/issues/1897
-  patchPhase = ''
-    for f in $(find . -iname "*.csproj"); do
-      sed -i '/Include="Microsoft.SourceLink.GitHub"/d' $f
-    done
-  '';
+  # Fixes application reporting 0.0.0.0 as its version.
+  MINVERVERSIONOVERRIDE = version;
 
-  buildPhase = ''
-    export FrameworkPathOverride=${mono}/lib/mono/4.7.1-api
+  dotnet-sdk = dotnetCorePackages.sdk_5_0;
+  dotnet-runtime = dotnetCorePackages.aspnetcore_5_0;
+
+  nativeBuildInputs = [ git glibcLocales bintools ];
+
+  runtimeDeps = [ mono ];
 
-    # disable default-source so nuget does not try to download from online-repo
-    nuget sources Disable -Name "nuget.org"
-    # add all dependencies to a source called 'nixos'
-    for package in ${toString deps}; do
-      nuget add $package -Source nixos
+  nugetBinariesToPatch = if stdenv.isLinux then [
+    "grpc.tools/2.39.1/tools/linux_x64/protoc"
+    "grpc.tools/2.39.1/tools/linux_x64/grpc_csharp_plugin"
+  ] else [ ];
+
+  postConfigure = ''
+    # Fixes execution of native protoc binaries during build
+    for binary in $nugetBinariesToPatch; do
+      path="$HOME/.nuget/packages/$binary"
+      echo "Patching $path"
+      ${patchelf}/bin/patchelf --set-interpreter "$(cat $NIX_BINTOOLS/nix-support/dynamic-linker)" $path
```

```suggestion
      patchelf --set-interpreter "$(cat $NIX_BINTOOLS/nix-support/dynamic-linker)" $path
```
patchelf is in path already.

---

Without it, the ofborg darwin build failed (due to not being able to find `patchelf`)

---

patchelf does not work on darwin IIRC.